lib/scanner/walk.go

```diff
@@ -274,17 +274,20 @@ func (w *walker) walkAndHashFiles(ctx context.Context, toHashChan chan<- protoco
 		// ignored path isn't actually a parent of the current path
 		if rel == path {
 			ignoredParent = ""
-			return w.handleItem(ctx, path, toHashChan, finishedChan, skip)
+			return w.handleItem(ctx, path, info, toHashChan, finishedChan, skip)
 		}
 
 		// The previously ignored parent directories of the current, not
 		// ignored path need to be handled as well.
-		if err = w.handleItem(ctx, ignoredParent, toHashChan, finishedChan, skip); err != nil {
-			return err
-		}
-		for _, name := range strings.Split(rel, string(fs.PathSeparator)) {
+		for _, name := range append([]string{""}, strings.Split(rel, string(fs.PathSeparator))...) {
```

Comment:
explain the change?

Reply:
That's the removed `handleItem` outside of the loop. It was a duplicate call before, and now it would be even more duplication (due to stating), thus I opted for the admittedly slightly hacky option of adding an empty string to handle the base path too. Added a comment.